### PR TITLE
Extract the registration request from the registration service

### LINKED ISSUE #244: Refactor RegistrationService#register_object
It's complexity is way too high.

### PR DESCRIPTION
**Pull request by jcoyne**
Fixes #244

### CODE REVIEW

File: app/models/registration_request.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    %i[object_type label admin_policy].each do |required_param|
+      raise Dor::ParameterError, "#{required_param.inspect} must be specified in call to register_object" unless params[required_param]
+    end
+    raise Dor::ParameterError, 'label cannot be empty to call register_object' if params[:label].empty? && %w[label none].include?(metadata_source)
+
+    raise Dor::ParameterError, "Unknown item type: '#{object_type}'" if item_class.nil?
+
+    raise Dor::ParameterError, "Unknown value for seed_datastream: '#{params[:seed_datastream]}'" if params[:seed_datastream] && params[:seed_datastream] != ['descMetadata']
+
+    raise ArgumentError, ":source_id Hash can contain at most 1 pair: recieved #{source_id.size}" if source_id.size > 1
```
Comment (justinlittman) [inline app/models/registration_request.rb]: Typo

Comment (justinlittman) [reply app/models/registration_request.rb]: Also, should this be a Dor::ParameterError.

File: app/models/registration_request.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    raise Dor::ParameterError, "Unknown rights setting '#{rights}' when calling #{name}.register_object" if rights && rights != 'default' && !Dor::RightsMetadataDS.valid_rights_type?(rights)
```
Comment (justinlittman) [inline app/models/registration_request.rb]: Confusing that rights isn't `param[:rights]`.

File: app/services/registration_service.rb [hunk 3]
```diff
@@ -1,10 +1,54 @@
...
-require 'uuidtools'
-
+# Registers an object. Creates a skeleton object in Fedora with a Druid.
 class RegistrationService
   class << self
-    # @TODO: Why isn't all this logic in, for example, Dor::Item.create? or Dor::Base.create? or Dor::Creatable.create?
+    # @param [Hash] params
+    # @see register_object similar but different
+    def create_from_request(params)
+      other_ids = Array(params[:other_id]).map do |id|
```
Comment (justinlittman) [inline app/services/registration_service.rb]: Move into separate method?

File: app/services/registration_service.rb [hunk 4]
```diff
@@ -1,10 +1,54 @@
...
+        if id =~ /^symphony:(.+)$/
+          "#{$1.length < 14 ? 'catkey' : 'barcode'}:#{$1}"
+        else
+          id
+        end
+      end
+
+      if params[:label] == ':auto'
```
Comment (justinlittman) [inline app/services/registration_service.rb]: Move into separate method?

File: app/services/registration_service.rb (hunk 3 above)
Comment (jcoyne) [reply app/services/registration_service.rb]: This isn't something I changed.  I just moved the methods position in the file.

File: app/services/registration_service.rb (hunk 4 above)
Comment (jcoyne) [reply app/services/registration_service.rb]: This isn't something I changed.  I just moved the methods position in the file.

File: app/services/registration_service.rb (hunk 3 above)
Comment (jcoyne) [reply app/services/registration_service.rb]: Was here: https://github.com/sul-dlss/dor-services-app/pull/245/files#diff-2c8dbe6f318ec983b990f295ec936436L131

File: app/services/registration_service.rb (hunk 4 above)
Comment (jcoyne) [reply app/services/registration_service.rb]: Was here: https://github.com/sul-dlss/dor-services-app/pull/245/files#diff-2c8dbe6f318ec983b990f295ec936436L139

File: app/models/registration_request.rb (hunk 1 above)
Comment (jcoyne) [reply app/models/registration_request.rb]: This is directly copied from https://github.com/sul-dlss/dor-services-app/pull/245/files#diff-2c8dbe6f318ec983b990f295ec936436L76 . The lack of changes was intentional.

**Pull request merged**